CRM/Campaign/BAO/Survey.php

```diff
@@ -949,27 +949,12 @@ public static function releaseRespondent($params) {
   }
 
   /**
-   * Get options for a given field.
-   * @see CRM_Core_DAO::buildOptions
-   *
-   * @param string $fieldName
-   * @param string $context : @see CRM_Core_DAO::buildOptionsContext
-   * @param array $props : whatever is known about this dao object
-   *
-   * @return array|bool
+   * Pseudoconstant condition_provider for activity_type_id field.
+   * @see \Civi\Schema\EntityMetadataBase::getConditionFromProvider
    */
-  public static function buildOptions($fieldName, $context = NULL, $props = []) {
-    $params = [];
-    // Special logic for fields whose options depend on context or properties
-    switch ($fieldName) {
-      case 'activity_type_id':
-        $campaignCompId = CRM_Core_Component::getComponentID('CiviCampaign');
-        if ($campaignCompId) {
-          $params['condition'] = ["component_id={$campaignCompId}"];
-        }
-        break;
-    }
-    return CRM_Core_PseudoConstant::get(__CLASS__, $fieldName, $params, $context);
+  public static function alterActivityTypes(string $fieldName, CRM_Utils_SQL_Select $conditions) {
```

Comment:
@colemanw did you miss to add the third fn argument $params here ?

Reply:
@monishdeb you're right that is passed to the callback, but in this case it's not used by the function so I just left it off.